nncf/experimental/torch/search_building_blocks/search_blocks.py

```diff
@@ -0,0 +1,694 @@
+from collections import deque
+from copy import deepcopy
+from enum import Enum
+from functools import cmp_to_key
+from itertools import combinations
+from itertools import groupby
+from typing import Any, List, Set, Tuple, Dict
+
+import networkx as nx
+
+import torch
+
+
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.graph.graph import NNCFNodeName
+from nncf.common.graph.graph_matching import find_subgraphs_matching_pattern
+from nncf.common.graph.definitions import MODEL_OUTPUT_OP_NAME
+from nncf.torch.dynamic_graph.operation_address import OperationAddress
+from nncf.torch.graph.graph import PTNNCFGraph
+from nncf.torch.hardware.fused_patterns import PT_HW_FUSED_PATTERNS
+from nncf.torch.graph.operator_metatypes import PTDropoutMetatype
+from nncf.torch.graph.operator_metatypes import PTRELUMetatype
+from nncf.torch.graph.operator_metatypes import PTMatMulMetatype
+from nncf.torch.graph.operator_metatypes import PTLinearMetatype
+from nncf.torch.layers import NNCF_MODULES_OP_NAMES
+from nncf.torch.nncf_network import NNCFNetwork
+
+
+IGNORED_NAME_OPERATORS = [*PTDropoutMetatype.get_all_aliases(), MODEL_OUTPUT_OP_NAME]
+
+
+class SearchGraphNode:
+
+    """
+    Class describing nodes used in SearchGraph.
+    """
+
+    def __init__(self,
+                 node_key: str,
+                 data: Dict = None):
+        self.node_key = node_key
+        self.data = data if data else {}
+
+    @property
+    def is_merged(self) -> bool:
+        """
+        Returns value that the node is merged.
+        """
+        return self.data.get(SearchGraph.IS_MERGED_NODE_ATTR)
+
+    @property
+    def is_dummy(self) -> bool:
+        """
+        Returns value that the node is dummy.
+        """
+        return self.data.get(SearchGraph.IS_DUMMY_NODE_ATTR)
+
+    @property
+    def node_name(self) -> NNCFNodeName:
+        return self.data.get(NNCFGraph.NODE_NAME_ATTR)
+
+    @property
+    def node_type(self) -> str:
+        """
+        Returns type of node.
+        """
+        return self.data.get(NNCFGraph.NODE_TYPE_ATTR)
+
+    @property
+    def layer_name(self) -> str:
+        """
+        Returns the name of the layer to which the node corresponds.
+        """
+        return self.data.get(NNCFGraph.LAYER_NAME_ATTR)
+
+    @property
+    def main_id(self) -> int:
+        """
+        Returns the id of node. In case if the node is merged returns id of the first node from merged node list.
+        """
+        # return the first id node
```

Comment:
Do you really need this comment if you have docstring?

Reply:
Fixed. Thanks!